menas/src/main/scala/za/co/absa/enceladus/menas/services/SchemaRegistryService.scala

```diff
@@ -61,6 +64,48 @@ class SchemaRegistryService @Autowired()() {
     }
   }
 
+  /**
+   * JKS based TrustManagerFactory option based on [[SecureConfig.Keys.javaxNetSslTrustStore]] and
+   * [[SecureConfig.Keys.javaxNetSslTrustStorePassword]] presence in [[config]]
+   */
+  lazy val trustManagerFactory: Option[TrustManagerFactory] = {
+    if (SecureConfig.hasTrustStoreProperties(config)) {
+      val trustStore = KeyStore.getInstance(defaultStoreType)
+
+      val tsInputStream = new FileInputStream(config.getString(SecureConfig.Keys.javaxNetSslTrustStore))
+      trustStore.load(tsInputStream, config.getString(SecureConfig.Keys.javaxNetSslTrustStorePassword).toCharArray)
+      tsInputStream.close()
+
+      val tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
+      tmf.init(trustStore)
+
+      Some(tmf)
+    } else {
+      None
+    }
+  }
+
+  /**
+   * JKS based KeyManagerFactory option based on [[SecureConfig.Keys.javaxNetSslKeyStore]] and
+   * [[SecureConfig.Keys.javaxNetSslKeyStorePassword]] presence in [[config]]
+   */
+  lazy val keyManagerFactory: Option[KeyManagerFactory] = {
+    if (SecureConfig.hasKeyStoreProperties(config)) {
+      val ks = KeyStore.getInstance(defaultStoreType)
+
+      val ksInputStream = new FileInputStream(config.getString(SecureConfig.Keys.javaxNetSslKeyStore))
```

Comment:
Is this safe? What if it does not exist?

Reply:
`SecureConfig.hasKeyStoreProperties(config)` should guard you for this very case.